devtools/etdump/etdump_filter.cpp

```diff
@@ -0,0 +1,100 @@
+/*
```

Comment:
Overall high level comment, i think it's reasonable that we can assume the etdump filter we provide based on regex will need malloc to be supported. If users want to use this malloc support will be required on their platform. Based on this i think it would be good to remove the limitations we have in this class such as max regex, max char len etc.

Reply:
If users do need a filter for their platform that isn't based on regex they can write their own one. (which will be only very few embedded users).

Reply:
> which will be only very few embedded users
ok  in this case i agree we can make our etdump_filter support regex + malloc, just as what [swolchok](https://github.com/swolchok) mentioned RE use malloc internally. Let me make an update